secrets/src/test/java/com/githhub/awsjavakit/secrets/SecretsReaderTest.java

```diff
@@ -0,0 +1,147 @@
+package com.githhub.awsjavakit.secrets;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNot.not;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.mockito.invocation.InvocationOnMock;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+
+class SecretsReaderTest {
+
+  static final String SECRET_VALUE = "SECRET_VALUE";
+  static final String SECRET_KEY = "SECRET_KEY";
+  static final String SECRET_NAME = "SECRET_NAME";
+  static final String PLAIN_TEXT_SECRET_NAME = "PLAIN_TEXT_SECRET_NAME";
+  static final String PLAIN_TEXT_SECRET_VALUE = "PLAIN_TEXT_SECRET_VALUE";
+  static final String JSON_SECRET_NAME = "JSON_SECRET_NAME";
+  static final String JSON_SECRET_VALUE = "{\"username\":\"name\", \"password\":\"pass\"}";
+  static final String JSON_LIST_SECRET_NAME = "JSON_LIST_SECRET_NAME";
+  static final String JSON_LIST_SECRET_VALUE = "[{\"key1\":\"val1\"},{\"key1\":\"val2\"}]";
+  static final String JSON_SECRET_VALUE_USERNAME = "name";
+  static final String JSON_SECRET_VALUE_PASSWORD = "pass";
+  static final String WRONG_SECRET_NAME = "WRONG_SECRET_NAME";
+  static final String WRONG_SECRET_KEY = "WRONG_KEY";
+  private static final ObjectMapper JSON = new ObjectMapper();
+  private final SecretsReader secretsReader;
+
+  SecretsReaderTest() {
+    secretsReader = createSecretsReaderMock();
+  }
+
+  @Test
```

Comment:
Here we have a major change in functionality. In NVA we were hiding which secret we were trying to read. Here, however the exception reveals the sercret name (but not the value). The secret name exists as a parameter (not a secret parameter) so in a way the secret name is not a secret in itself. If however the client needs to hide the secret name, they can do so by catching the exception.